### DISCUSSION TITLE
"View only shared album" [Feature]:

### DISCUSSION THREAD

**Discussion by Morethanevil** [immich-app/immich#1607] [Feature Request]
### Feature detail

It is now possible for public shares to toggle upload on and off. 
I would like this for internal shares with other user too. 
By default, if I create a shared internal album, the other user may upload in them. But what if I only want them to view images? I could create a link, but then everyone with the link has access. 

Another nice to have would be an indicator on a shared album, if somebody uploads in a share. 

### Platform

Web

**Comment by jrasm91** [comment]
I think this is being tracked in a more broad issue related to users, groups, and permissions. Basically once users can be assigned roles and levels of access this can be expanded to albums as well.

**Comment by SquirmDog** [comment]
It would be nice customize each users abilities on the album. For example:

- User 1 is read only, can't add and cannot delete
- User 2 could add but cannot delete
- User 3 could both add and delete

**Comment by mgabor3141** [comment]
I'm looking to contribute this feature. My plan is the following:

Sharing an album with a user would add that user as a Viewer. They would not be able to upload images or make any modifications to the album. This would behave a little bit like the link sharing, but the user would still need to be logged in. These albums would also show up in the users shared album list.

The album sharing modal would be modified like so:

![image](https://github.com/immich-app/immich/assets/9047995/63553e79-db7d-4667-8747-e74ff6ad8941)

A DB column would be added for this sharing type, and backend endpoints would be modified to support this functionality.
